tests/test_peak_processing.py

```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
+       st.integers(min_value=1900, max_value=10000),
+       st.integers(min_value=1000, max_value=int(7_000_000)),
+       )
+@example(
+    records=np.array([(0, 1, 1, 0, 0, 0, 0, 0, 0., 0., 0, [1, 0]),
+                      (1, 1, 1, 1, 0, 0, 0, 0, 0., 0., 0, [1, 0])],
+                     dtype=strax.record_dtype(2)),
+    gap_factor=108,
+    max_duration=1000,
+    right_extension=5000,
+    gap_threshold=18000,
+)
+def test_peak_overflow(records,
+                       gap_factor,
+                       right_extension,
+                       gap_threshold,
+                       max_duration,
+                       ):
+
+    """
+    Test that we handle dt overflows in peaks correctly. To this end, we
+        just create some sets of records and copy that set of records
+        for a few times. That way we may end up with a very long
+        artificial set of hits that can be used in the peak building. By
+        setting the peak finding parameters to very strange conditions
+        we are able to replicate the behaviour where a peak would become
+        so large that it cannot be written out correctly due to integer
+        overflow of the dt field,
+    :param records: records
+    :param gap_factor: to create very extended sets of records, just
+        add a factor that can be used to multiply the time field with,
+        to more quickly arrive to a very long pulse-train
+    :param max_duration: max_duration option for strax.find_peaks
+    :param right_extension: option for strax.find_peaks
+    :param gap_threshold: option for strax.find_peaks
+    :return: None
+    """
+
+    # Set this here, no need to test left and right independently
+    left_extension = 0
+    p = np.zeros(0, dtype=strax.peak_dtype())
+    magic_overflow_time = np.iinfo(p.dtype['dt']).max * p.dtype['data'].shape[0]
+    del p
+
+    # Make a single big peak to contain all the records
+    def retrun_1(x):
+        """
+        Return 1 for all of the input that can be used as a parameter
+            for the splitting in natural breaks
+        :param x: any type of array
+        :return: ones * len(array)
+        """
+        ret = np.ones(len(x))
+        return ret
+
+    r = records
+    if not len(r) or len(r['channel']) == 1:
+        # Hard to test integer overflow for empty records or with
+        # records only from a single channel
+        return
+
+    # Copy the pulse train of the records. We are going to copy the same
+    # set of records many times now
+    r_buffer = []
+    t_max = strax.endtime(r)
+    max_record_repetition_factor = 1_000_000
+    for i in range(max_record_repetition_factor):
+        r_copy = r.copy()
+        r_copy['time'] = r_copy['time'] + t_max * i * gap_factor
+        r_buffer.append(r_copy)
+        if r_copy['time'][-1] - r['time'][0] > 1.5 * magic_overflow_time:
+            # No need to go over and beyond. We should get int-overflow
+            break
+    r = np.concatenate(r_buffer)
```

Comment:
This is a bit confusing to me-. Are not you just creating equidistant intervals like: 
```
time = np.arange(time, 1,5* magic_overflow_time + t_max*gap_factor, t_max*gap_factor)
```
which would be much simpler?

Reply:
indeed, this is exactly what it is doing but I also want to have some records data, if it were just the time fields I was interested in, you would be right.

Reply:
But the data stays the same is not it. It is something like result['data'][:] = r['data']

Reply:
okay, yeah it is a bit convoluted, I've changed to `np.tile` to do almost exactly the same, thanks for the suggestion